### PR TITLE
Convert all daemons to use entrypoints

### PR DESCRIPTION
**Pull request by mbland**
Depends on #6. I may push more tweaks while that PR is still pending, so I'll ping when ready.

cc: @jcscottiii @ccostino @afeld @catherinedevlin

### CODE REVIEW

**Comment by mbland** [comment]
cc: @batemapf

File: authdelegate/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#! /bin/bash -e
+
+source /etc/profile.d/gvm.sh
+
+if [ "$1" = "run-server" ]; then
```
Comment (jcscottiii) [inline authdelegate/entrypoint.sh:5]: For this entrypoint script and the others, will you ever not pass in `run-server` as a parameter? (just going solely based on what I see passed in the `CMD` in the Dockerfiles. So I could be missing another purpose of the entrypoint outside of the Dockerfile.)

**Comment by jcscottiii** [comment]
just had one question above @mbland. looks good!

File: authdelegate/entrypoint.sh (hunk 1 above)
Comment (mbland) [inline authdelegate/entrypoint.sh:5]: Yes; when running `./go run_container nginx /bin/bash` or some such to set up the environment but drop into a shell to poke around without running the service. Also, in #8, I found out how to use the same `hmacproxy` image to run an auth server and a signing proxy in two different containers, depending on if it's started with `run-server` or `run-proxy`.

I picked up the idea from https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#entrypoint.

**Pull request merged**